version: '3.8'

services:
  nginx:
    image: vvooki/moja-aplikacja-nginx:latest
    configs:
      - source: custom_nginx_conf
        target: /etc/nginx/nginx.conf
    deploy:
      replicas: 2
      restart_policy:
        condition: on-failure
      labels: 
        - "traefik.enable=true"
        - "traefik.http.routers.nginx.rule=Host(`lukasz.vps1.m89.pl`)" 
        - "traefik.http.routers.nginx.entrypoints=web"
        - "traefik.http.services.nginx.loadbalancer.server.port=80"
    networks:
      - web

  php:
    image: vvooki/moja-aplikacja-php:latest
    deploy:
      replicas: 1
    networks:
      - web

  db:
    image: vvooki/moja-aplikacja-sql:latest
    environment:
      MYSQL_ROOT_PASSWORD: rootpass
      MYSQL_DATABASE: testdb
      MYSQL_USER: testuser
      MYSQL_PASSWORD: testpass
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      timeout: 5s
      retries: 5
    deploy:
      replicas: 1
    networks:
      - web

configs:
  custom_nginx_conf:
    file: ./nginx/nginx-swarm.conf

networks:
  web:
    external: true
